apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,269 @@
+ * Determines if file generation should be skipped based on templateConfig conditions.
+ *
+ * @param  {AsyncAPIDocument} asyncapiDocument AsyncAPI document to use for condition evaluation.
+ * @param  {String} relativeSourceFile Relative path of the source file.
+ * @param  {String} relativeSourceDirectory Relative path of the source directory.
+ * @param  {String} relativeTargetFile Relative path of the target file to be generated.
+ * @param  {Object} templateConfig Template configuration.
+ * @param  {String} targetDir Directory where the generated files are written.
+ * @return {Promise<Boolean>} Returns true if generation should be skipped.
+ */
+async function conditionalGeneration (
+  relativeSourceFile,
+  relativeSourceDirectory,
+  relativeTargetFile,
+  templateConfig,
+  targetDir,
+  matchedConditionPath,
+  templateParams,
+  asyncapiDocument 
+) {
+  const conditionFilesGeneration = templateConfig?.conditionalFiles?.[matchedConditionPath] || {};
+  const conditionalGeneration = templateConfig?.conditionalGeneration?.[matchedConditionPath] || {};
+
+  const config = Object.keys(conditionFilesGeneration).length > 0 
+    ? conditionFilesGeneration 
+    : conditionalGeneration;
+  
+  const parameter = config?.parameter;
+  const subject = config?.subject;
+  const validation = config?.validation;
+  if (Object.keys(conditionFilesGeneration).length>0 && subject) {
+    return conditionalFilesGenerationDeprecateVersion(
+      asyncapiDocument,
+      templateParams,
+      templateConfig,
+      relativeSourceFile
+    );
+  } else 
+  if (Object.keys(conditionalGeneration).length>0 && subject) {
+    return conditionalSubjectGeneration(asyncapiDocument,templateConfig,matchedConditionPath);
+  }
+  const parameterValue = getParameterValue(templateParams, parameter);
+
+  if (parameterValue === undefined) {
+    handleMissingParameterValue(matchedConditionPath, templateConfig);
+    return false;
+  }
+
+  return validateParameterValue(
+    validation,
+    parameterValue,
+    matchedConditionPath,
+    relativeSourceDirectory,
+    relativeTargetFile,
+    targetDir
+  );
+};
+
+/**
+ * Determines whether a file should be conditionally included based on the provided subject expression
+ * and optional validation logic defined in the template configuration.
+ *
+ * @param {Object} asyncapiDocument - The parsed AsyncAPI document instance used for context evaluation.
+ * @param {Object} templateParams - The template parameters passed by the user (e.g. server selection).
+ * @param {Object} templateConfig - The configuration object that contains `conditionalFiles` rules.
+ * @param {String} relativeSourceFile - The relative path to the source file being evaluated.
+ * @param {String} relativeSourceDirectory - The relative path to the directory of the source file.
+ * @returns {Boolean} - Returns `true` if the file should be included; `false` if it should be skipped.
+ */
+async function conditionalFilesGenerationDeprecateVersion (
```

Comment:
```suggestion
async function conditionalFilesGenerationDeprecatedVersion (
```

or just `legacyConditionalFilesGeneration`